cds_migrator_kit/rdm/migration/transform/xml_processing/rules/base.py

```diff
@@ -71,12 +71,85 @@ def title(self, key, value):
     return value.get("a", "TODO")
 
 
+abbreviations = []
+
+
+@model.over("description", "^246__")
+def description(self, key, value):
+    """Translates description."""
+    is_abbreviation = value.get("i") == "Abbreviation"
+    abbreviations.append(value.get("a"))
+
+    if is_abbreviation:
+        return "Abbreviations: " + "; ".join(abbreviations)
+
+    return
+
+
+@model.over("additional_descriptions", "(^500__)|(^520__)")
+@for_each_value
+@require(["a"])
+def additional_descriptions(self, key, value):
+    """Translates additional description."""
+    description_text = value.get("a")
+
+    if key == "500__":
+        additional_description = {
+            "description": description_text,
+            "type": {
+                "id": "other",  # what's with the lang
+            }
+        }
+    elif key == "520__":
+        additional_description = {
+            "description": description_text,
+            "type": {
+                "id": "abstract",  # what's with the lang
+            }
+        }
+
+    return additional_description
+
+
+@model.over("imprint", "^269__")
+def imprint(self, key, value):
+    """Translates imprint - WARNING - also publisher and publication_date."""
+
+    imprint = {
+        "place": value.get("a"),
+    }
+
+    if not self.get("publication_date"):
```

Comment:
this is tricky, you assume that some other field takes precedence. What is the field? In which circumstances can we get the value already assigned to that key before we analyse 269?